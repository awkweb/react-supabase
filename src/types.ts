import { PostgrestFilterBuilder } from '@supabase/postgrest-js'

export type Filter<Data> = (
    query: PostgrestFilterBuilder<Data>,
) => PostgrestFilterBuilder<Data>

export type PostgrestError = {
    message: string
    details: string
    hint: string
    code: string
}
