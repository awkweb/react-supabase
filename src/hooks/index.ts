export * from './use-client'
export * from './use-filter'
export * from './use-select'
