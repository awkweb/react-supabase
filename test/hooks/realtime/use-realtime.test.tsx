import { renderHook } from '@testing-library/react-hooks'

import { useRealtime } from '../../../src'
import { Wrapper as wrapper } from '../../utils'

describe('useRealtime', () => {
    it('should throw when not inside Provider', () => {
        const { result } = renderHook(() => useRealtime('todos'))
        expect(() => result.current).toThrowErrorMatchingSnapshot()
    })

    it('should throw when trying to listen all database changes', () => {
        const { result } = renderHook(() => useRealtime('*'), { wrapper })
        expect(() => result.current).toThrowErrorMatchingSnapshot()
    })

    it('should throw when trying to listen all database changes via options', () => {
        const { result } = renderHook(() => useRealtime({ table: '*' }), {
            wrapper,
        })
        expect(() => result.current).toThrowErrorMatchingSnapshot()
    })
})
