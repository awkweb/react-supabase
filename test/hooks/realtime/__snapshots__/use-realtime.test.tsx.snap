// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRealtime should throw when not inside Provider 1`] = `"No client has been specified using Provider."`;

exports[`useRealtime should throw when trying to listen all database changes 1`] = `"Must specify table or row. Cannot listen for all database changes."`;

exports[`useRealtime should throw when trying to listen all database changes via options 1`] = `"Must specify table or row. Cannot listen for all database changes."`;
